src/core/metrics.cc

```diff
@@ -129,10 +132,26 @@ Metrics::HashLabels(const std::map<std::string, std::string>& labels)
 Metrics::~Metrics()
 {
 #ifdef TRITON_ENABLE_METRICS_GPU
-  // Signal the nvml thread to exit and then wait for it...
-  if (nvml_thread_ != nullptr) {
-    nvml_thread_exit_.store(true);
-    nvml_thread_->join();
+  // Signal the dcgm thread to exit and then wait for it...
+  if (dcgm_thread_ != nullptr) {
+    dcgm_thread_exit_.store(true);
+    dcgm_thread_->join();
+    dcgmGroupDestroy(dcgm_handle_, groupId_);
+    // Stop and shutdown DCGM
+    dcgmReturn_t derr;
+    if (standalone_) {
+      derr = dcgmDisconnect(dcgm_handle_);
+    } else {
+      derr = dcgmStopEmbedded(dcgm_handle_);
+    }
+    if (derr != DCGM_ST_OK) {
+      LOG_WARNING << "error, unable to stop DCGM: " << errorString(derr);
```

Comment:
Here and elsewhere. Don't say  "error" when it is a warning. Experience shows that we will bet issues and bugs filed saying an error happened. just put "unable to stop... ".